src/platform/qpg6100/qpg6100Config.cpp

```diff
@@ -284,6 +246,33 @@ CHIP_ERROR QPG6100Config::ForEachRecord(Key firstKey, Key lastKey, bool addNewRe
     return err;
 }
 
+CHIP_ERROR QPG6100Config::MapNVMError(qvStatus_t aStatus)
+{
+    CHIP_ERROR err;
+
+    switch (aStatus)
+    {
+    case QV_STATUS_NO_ERROR:
+        err = CHIP_NO_ERROR;
+        break;
+    case QV_STATUS_BUFFER_TOO_SMALL:
+        err = CHIP_ERROR_BUFFER_TOO_SMALL;
+        break;
+    case QV_STATUS_INVALID_ARGUMENT:
+        err = CHIP_ERROR_INVALID_ARGUMENT;
+        break;
+    case QV_STATUS_KEY_LEN_TOO_SMALL:
+        err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
+        break;
+    case QV_STATUS_INVALID_DATA:
+        err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
```

Comment:
Should we be mixing `CHIP_ERROR` and `CHIP_DEVICE_ERROR`? There is `CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND` that could work here.